src/memory_server.ml

```diff
@@ -21,10 +21,31 @@ open Squeezed_state
 open Squeezed_xenstore
 open Xapi_stdext_monadic
 open Xapi_stdext_unix
+open Xapi_stdext_std
+open Xapi_stdext_threads
 
 module D = Debug.Make(struct let name = Memory_interface.service_name end)
 open D
 
+module Host_free_memory_event = struct
+  let accesible = ref false
```

Comment:
Why do we need this? Isn't it enough to wait on the condition?

Reply:
If we do not have this variable, then we may run into following scenario:
1. Thread-1 set the condition
2. Then Thread-2 wait on the condition (and it will not be waked up anymore)

So In Thread-2 we have to check the variable before waiting on the condition.

Reply:
Alternatively, we can use the presence of the tmp file to represent this Boolean. If the file exists, then the caller can just read it and return without waiting. This also ensures that we won't introduce unnecessary waiting when restarting squeezed. In fact, it would be wrong to recalculate the boot-time free memory when restarting the service, without a host reboot.

Reply:
The file exists when the writer begin writing, so we may read an incomplete content of the file. Otherwise we should have a lock to protect the read and write.

Reply:
Or write to another file and do an atomic move.

Reply:
This can be done with https://github.com/xapi-project/stdext/blob/master/lib/xapi-stdext-unix/unixext.mli#L68

Reply:
Yes, but keep the variable here could made the logic simplier. I was inspired by the python `Event` class to create this module.

Reply:
I still want to ensure that we don't wait and rewrite the file on restart, if the file already exists.

Reply:
Yes, the current implementation already ensure that.